contracts/ComptrollerStorage.sol

```diff
@@ -104,10 +104,34 @@ contract ComptrollerStorage {
 
     // No collateralFactorMantissa may exceed this value
     uint256 internal constant collateralFactorMaxMantissa = 0.9e18; // 0.9
+
+    /// @notice Indicator that this is a Comptroller contract (for inspection)
+    bool internal constant _isComptroller = true;
+
+    // PoolRegistry, immutable to save on gas
+    /// @custom:oz-upgrades-unsafe-allow state-variable-immutable
+    address public immutable poolRegistry;
+
+    // AccessControlManager, immutable to save on gas
+    /// @custom:oz-upgrades-unsafe-allow state-variable-immutable
+    address public immutable accessControl;
+
     /**
      * @dev This empty reserved space is put in place to allow future versions to add new
      * variables without shifting down storage in the inheritance chain.
      * See https://docs.openzeppelin.com/contracts/4.x/upgradeable#storage_gaps
      */
     uint256[50] private __gap;
+
+    /// @custom:oz-upgrades-unsafe-allow constructor
+    constructor(address poolRegistry_, address accessControl_) {
```

Comment:
@defcon022 Shouldn't the constructor be in `Comptroller` contract instead of `ComptrollerStorage`?

Reply:
`ComptrollerStorage` is inherited in `Comptroller` and every storage variable is been shifted to `ComptrollerStorage` contract. 

The same issue been raised by Certik.